src/Provider/DebugServiceProvider.php

```diff
@@ -92,91 +93,47 @@ public function register(Application $app)
             });
         }
 
-        $app['debug.handlers_listener'] = $app->share(
-            function ($app) {
-                $errorLevels = $app['config']->get(
-                    $app['debug'] ?
-                    'general/debug_error_level' :
-                    'general/production_error_level'
-                );
-
-                return new DebugHandlersListener(
-                    null, // HttpKernel or Console App exception handler
-                    $app['logger'],
-                    $errorLevels, // Levels
-                    $errorLevels, // Throw at
-                    true, // Scream
-                    $app['code.file_link_format']
-                );
+        // Thrown errors in an integer bit field of E_* constants
+        $app['error_handler.throw_at'] = function ($app) {
+            if ($app['debug']) {
+                return $app['config']->get('general/debug_error_level', E_ALL);
+            } else {
+                return $app['config']->get('general/production_error_level', 0);
             }
-        );
+        };
+
+        // Logged errors in an integer bit field of E_* constants.
+        $app['error_handler.log_at'] = E_ALL; // TODO Should this be the same as throw_at? Previously it was via config env_error_level.
```

Comment:
`s/env_error_level/debug_error_level/`

Nitpick aside, and onto O.C.D., adding a TODO :cry: 

I think "thrown" and "logged" levels are somewhat related, but should be separable.

However, for example if I have my `debug_error_level` set to `E_ALL &~ E_DEPRECATED &~ E_USER_DEPRECATED`, aren't my (by default `$app['logger']`) log target going to potentially get flooded with deprecations, making their own errors harder to find?

tl;dr ` $app['error_handler.log_at']` is not invoked until `boot()` so probably should match config's `debug_error_level` for now … and **maybe** we later make it a separate config later?